src/common/lru_cache.h

```diff
@@ -453,6 +461,208 @@ std::shared_ptr<CacheMetrics>
     return  cacheMetrics_;
 }
 
+// TimedLRUCache
+template <typename K,  typename V,
+    typename KeyTraits = CacheTraits<K>,
+    typename ValueTraits = CacheTraits<V>>
+class TimedLRUCache : public LRUCacheInterface<K, V> {
+ public:
+    struct Item {
+        const K* key;
+        V value;
+        uint64_t time;
+    };
+
+ public:
+    explicit TimedLRUCache(uint64_t timeout,
+        std::shared_ptr<CacheMetrics> cacheMetrics = nullptr)
+      : timeout_(timeout),
+        maxCount_(0),
+        cacheMetrics_(cacheMetrics) {}
+
+    explicit TimedLRUCache(uint64_t timeout,
+        uint64_t maxCount,
+        std::shared_ptr<CacheMetrics> cacheMetrics = nullptr)
+      : timeout_(timeout),
+        maxCount_(maxCount),
+        cacheMetrics_(cacheMetrics) {}
+
+    void Put(const K &key, const V &value) override;
+
+    bool Put(const K &key, const V &value, V *eliminated) override;
+
+    bool Get(const K &key, V *value) override;
+
+    void Remove(const K &key) override;
+
+    uint64_t Size() override;
+
+    std::shared_ptr<CacheMetrics> GetCacheMetrics() const;
+
+ private:
+    bool PutLocked(const K &key, const V &value, V *eliminated);
+
+    void RemoveLocked(const K &key);
+
+    void MoveToFront(const typename std::list<Item>::iterator &elem);
+
+    bool RemoveOldest(V *eliminated);
+
+    void RemoveElement(const typename std::list<Item>::iterator &elem);
```

Comment:
There will be problems with directly Remove Element, it may be in a dirty state.

Reply:
Yes, so the iCache doesn't use TimedLRUCache and timeout is set in inodeWrapper to deal with dirty inode.
The TimeLRUCache is used for Dcache nowadays.